lib/ansible/modules/web_infrastructure/letsencrypt.py

```diff
@@ -332,7 +332,10 @@ def write_file(module, dest, content):
     checksum_dest = None
     # raise an error if there is no tmpsrc file
     if not os.path.exists(tmpsrc):
-        os.remove(tmpsrc)
+        try:
+            os.remove(tmpsrc)
```

Comment:
Why bother removing it in the first place? This exception should occur 100% of the time as long as `os` isn't bugged.

Reply:
It could be that the operating system returns strange results, on which `os` relies. If this happens, something really strange is going on, so trying again to delete doesn't hurt IMO.

But maybe such logic should be somewhere inside `module` itself, similar to `module.atomic_move()`, so no module developer who just wants to write a simple file has to think about this.